api.yml

```diff
@@ -0,0 +1,203 @@
+swagger: "2.0"
+
+info:
+  version: "0.1"
+  title: Kubernetes Queen API
+  description: A simple API to interact with Kubernetes clusters
+
+schemes:
+  - https
+  - http
+
+basePath: /api/v1
+
+definitions:
+  Cluster:
+    type: object
+    properties:
+      id:
+        type: "integer"
+        format: "int64"
+      name:
+        type: "string"
+      provisioner:
+        type: "integer"
+        format: "int64"
+      state:
+        type: "string"
+        enum:
+          - "deployed"
+          - "pending"
+          - "failed"
+          - "deleted"
+  ClusterNode:
+    type: object
+    properties:
+      name:
+        type: string
+      # TODO: more properties
+  ClusterService:
+    type: object
+    properties:
+      name:
+        type: string
+      # TODO: more properties
+  Provisioner:
+    type: object
+    properties:
+      id:
+        type: "integer"
```

Comment:
What does this id means?

Reply:
unique identifier of that object, or we can use names as unique identifiers and don't use ids at all

Reply:
ID and name aren't the same - if you delete object with some name and than create new one with same name, the ID should be changed, so, you can differentiate the objects.